### PR TITLE
ui/hooks: UserContactMethodEditDialog

### PR DESCRIPTION
**Pull request by Voninkazo**
- [x] Identified the issue which this PR solves.
- [ ] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [ ] Code builds clean without any errors or warnings.
- [ ] Added appropriate tests for any new functionality.
- [ ] All new and existing tests passed.
- [ ] Added comments in the code, where necessary.
- [ ] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Refactoring UserContactMethodEditDialog class component to a functional component and removing unused code

**Which issue(s) this PR fixes:**
Part of #923

### CODE REVIEW

**Review by KatieMSB — APPROVED** [review]
Looks good

File: web/src/app/users/UserContactMethodEditDialog.js [hunk 1]
```diff
@@ -25,57 +25,30 @@ const mutation = gql`
...
-        }
-      </Mutation>
-    )
-  }
+export default function UserContactMethodEditDialog({
+  onClose,
+  contactMethodID,
+}) {
+  const [value, setValue] = useState(null)
+  const [edit, setEdit] = useState(true)
```
Comment (KatieMSB) [inline web/src/app/users/UserContactMethodEditDialog.js:33]:
```suggestion
  const [edit] = useState(true)
```

**Review by KatieMSB — CHANGES_REQUESTED** [review]

File: web/src/app/users/UserContactMethodEditDialog.js (hunk 1 above)
Comment (Forfold) [reply web/src/app/users/UserContactMethodEditDialog.js:33]: It doesn't look like this is used anywhere other than passing as a prop, so we can probably just delete this line and do that directly below, on line 61:
```
edit={true}
```

File: web/src/app/users/UserContactMethodEditDialog.js [hunk 2]
```diff
@@ -57,7 +56,7 @@ export default function UserContactMethodEditDialog({
           <UserContactMethodForm
             errors={fieldErrs}
             disabled={loading}
-            edit={edit}
+            edit={true}
```
Comment (Forfold) [inline web/src/app/users/UserContactMethodEditDialog.js:59]: whoops, sorry for that suggestion, with the linter it should read as
```suggestion
            edit
```

**Review by Forfold — APPROVED** [review]

**Review by KatieMSB — APPROVED** [review]

**Pull request merged**